### PR TITLE
Add grpc plugin directive

### PR DESCRIPTION
**Pull request by pieterlouw**
### 1. What does this change do, exactly?
New directive for the http server type called `grpc`

### 2. Please link to the relevant issues.
N/A

### 3. Which documentation changes (if any) need to be made because of this PR?
No document changes needed. Link to plugin's documentation --> [here](https://github.com/pieterlouw/caddy-grpc/blob/master/README.md)

### CODE REVIEW

File: caddyhttp/httpserver/plugin.go [hunk 1]
```diff
@@ -490,6 +490,8 @@ var directives = []string{
 	"hugo",      // github.com/hacdias/caddy-hugo
 	"mailout",   // github.com/SchumacherFM/mailout
 	"awslambda", // github.com/coopernurse/caddy-awslambda
+
```
Comment (mholt) [inline caddyhttp/httpserver/plugin.go]: Let's take out this extra line.

Comment (pieterlouw) [inline caddyhttp/httpserver/plugin.go]: What is the best way to determine where in the chain are the best place to put the directive?

Comment (mholt) [inline caddyhttp/httpserver/plugin.go]: Where ever it works correctly and doesn't break other middlewares in the chain. We can always move it later, but I always ask to double check and make sure that the author at least considered that the plugin is in the right place. Whatever makes the most logical sense in the order of things done by handlers, really.

Comment (pieterlouw) [inline caddyhttp/httpserver/plugin.go]: Okay. I have left it at the bottom of the chain for now. I tested it along with tls,log and prometheus plugins on my dev machine and it all works fine

**Comment by mholt** [comment]
Cool, thanks. Just fix the spacing with the comment (gofmt -s) and I'll merge this in! (Check CI failures)

**Comment by mholt** [comment]
@pieterlouw You may now deploy your plugin to the Caddy website!

**Comment by pieterlouw** [comment]
Thanks @mholt! I just registered it now.

What's the steps if there's a new version of the plugin? Redeploy and then users need to build a new download of Caddy?

**Comment by mholt** [comment]
@pieterlouw Yep.

**Pull request merged**